File: elixir-ms/lib/domain/use_cases/cases_use_case.ex
defmodule ElixirMs.Domain.UseCase.CasesUseCase do
  alias ElixirMs.Domain.Model.Primes

  @hello_gateway Application.compile_env(:elixir_ms, :hello_gateway)

  def case_one(latency) do
    with {:ok, _} <- @hello_gateway.hello(latency),
         primes <- Primes.get_primes_list(1000) do
      {:ok, primes}
    end
  end

  def case_two(latency) do
    @hello_gateway.hello(latency)
  end

  def case_three() do
    Primes.get_primes_list(1000)
  end
end

File: java-helidon-nima-ms/nima/src/main/java/com/maocq/helidon/nima/domain/usecases/cases/CasesUseCase.java
package com.maocq.helidon.nima.domain.usecases.cases;

import com.maocq.helidon.nima.domain.model.hello.gateways.HelloGateway;
import com.maocq.helidon.nima.domain.model.primes.Primes;

public class CasesUseCase {
    private final HelloGateway helloGateway;

    public CasesUseCase(HelloGateway helloGateway) {
        this.helloGateway = helloGateway;
    }

    public String caseOne(int latency) {
        var hello = helloGateway.hello(latency);
        return Primes.primes(1000);
    }

    public String caseTwo(int latency) {
        return helloGateway.hello(latency);
    }

    public String caseThree(int n) {
        return Primes.primes(n);
    }
}
